nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITListAzureDataLakeStorage.java

```diff
@@ -0,0 +1,235 @@
+
+        runProcessor();
+
+        assertSuccess();
+    }
+
+    @Test
+    public void testListNonExistingDirectory() throws Exception {
+        runner.setProperty(AbstractAzureDataLakeStorageProcessor.DIRECTORY, "dummy");
+
+        runProcessor();
+
+        assertFailure();
+    }
+
+    @Test
+    public void testListWithNonExistingFileSystem() {
+        runner.setProperty(AbstractAzureDataLakeStorageProcessor.FILESYSTEM, "dummy");
+        runner.setProperty(AbstractAzureDataLakeStorageProcessor.DIRECTORY, "");
+
+        runProcessor();
+
+        assertFailure();
+    }
+
+    @Test
+    public void testListWithRecords() throws InitializationException {
+        runner.setProperty(AbstractAzureDataLakeStorageProcessor.DIRECTORY, "dir1");
+
+        MockRecordWriter recordWriter = new MockRecordWriter(null, false);
+        runner.addControllerService("record-writer", recordWriter);
+        runner.enableControllerService(recordWriter);
+        runner.setProperty(ListAzureDataLakeStorage.RECORD_WRITER, "record-writer");
+
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ListAzureDataLakeStorage.REL_SUCCESS, 1);
+        MockFlowFile flowFile = runner.getFlowFilesForRelationship(ListAzureDataLakeStorage.REL_SUCCESS).get(0);
+        flowFile.assertAttributeEquals("record.count", "3");
+    }
+
+    private void runProcessor() {
+        runner.assertValid();
+        runner.run();
+    }
+
+    private void assertSuccess(String... testFilePaths) throws Exception {
+        runner.assertTransferCount(ListAzureDataLakeStorage.REL_SUCCESS, testFilePaths.length);
+
+        Map<String, TestFile> expectedFiles = new HashMap<>(testFiles);
+        expectedFiles.keySet().retainAll(Arrays.asList(testFilePaths));
+
+        List<MockFlowFile> flowFiles = runner.getFlowFilesForRelationship(ListAzureDataLakeStorage.REL_SUCCESS);
+
+        for (MockFlowFile flowFile : flowFiles) {
+            String filePath = flowFile.getAttribute("azure.filePath");
+            TestFile testFile = expectedFiles.remove(filePath);
+            assertNotNull("File path not found in the expected map", testFile);
+            assertFlowFile(testFile, flowFile);
+        }
+    }
+
+    private void assertFlowFile(TestFile testFile, MockFlowFile flowFile) throws Exception {
+        flowFile.assertAttributeEquals("azure.filesystem", fileSystemName);
+        flowFile.assertAttributeEquals("azure.filePath", testFile.getFilePath());
+        flowFile.assertAttributeEquals("azure.directory", testFile.getDirectory());
+        flowFile.assertAttributeEquals("azure.filename", testFile.getFilename());
+        flowFile.assertAttributeEquals("azure.length", String.valueOf(testFile.getFileContent().length()));
+
+        flowFile.assertAttributeExists("azure.lastModified");
+        flowFile.assertAttributeExists("azure.etag");
+
+        flowFile.assertContentEquals("");
+    }
+
+    private void assertFailure() {
```

Comment:
`assertFailure` is misleading imo (as there is no `failure` relationship).
Could call `assertNoop` or `assertIgnore` instead.

Reply:
It is a failure case without sending FF to failure (List processors typically have only success and do not send FF on error).
Added assert for checking the error logging.